systemd-swap

```diff
@@ -70,7 +70,16 @@ get_fs_type(){
   elif [ -d $(dirname "$1") ]; then path=$(dirname "$1");
   else ERRO "swapfc_path is invalid"
   fi
-  df "$path" --output=fstype | tail -n 1
+
+  fstype=$(df "$path" --output=fstype | tail -n 1)
+
+  if [ "$fstype" == "-" ] && btrfs subvolume show "$path" &>/dev/null; then
+    fstype="btrfs"
+  else
+    ERRO '$swapfc_path is located on an unknown filesystem'
+  fi
+
+  echo "$fstype"
```

Comment:
Why echo $fstype?

Reply:
How else would you return the value of $fstype in bash? Before df echo'ed the result but now we have to do it ourselves since we intercepted df's output. Or am I wrong?

Reply:
Ahhh, you're quite right